test/integration/controller_test.go

```diff
@@ -575,6 +575,185 @@ func TestProvisionFailure(t *testing.T) {
 	}
 }
 
+func TestBindingFailure(t *testing.T) {
+	_, fakeCatalogClient, _, _, _, shutdownServer := newTestController(t, fakeosb.FakeClientConfiguration{
+		CatalogReaction: &fakeosb.CatalogReaction{
+			Response: &osb.CatalogResponse{
+				Services: []osb.Service{
+					{
+						Name:        testServiceClassName,
+						ID:          "12345",
+						Description: "a test service",
+						Bindable:    true,
+						Plans: []osb.Plan{
+							{
+								Name:        testPlanName,
+								Free:        truePtr(),
+								ID:          "34567",
+								Description: "a test plan",
+							},
+						},
+					},
+				},
+			},
+		},
+		BindReaction: &fakeosb.BindReaction{
+			Error: osb.HTTPStatusCodeError{
+				StatusCode:   http.StatusConflict,
+				ErrorMessage: strPtr("ServiceBindingExists"),
+				Description:  strPtr("Service binding with the same id, for the same service instance already exists."),
+			},
+		},
+		UnbindReaction: &fakeosb.UnbindReaction{},
+		ProvisionReaction: &fakeosb.ProvisionReaction{
+			// Async is temporarily disabled due to a race condition where
```

Comment:
In order to test the async flow for this, you'll need to do a synchronous provision, since you can only control a single async response in the simple fake we're currently using.